### PR TITLE
Add v2 buf.yaml LintWithWorkspaceTest test cases

### PR DESCRIPTION
**Pull request by drice-buf**

### CODE REVIEW

File: src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt [hunk 1]
```diff
@@ -21,4 +21,34 @@ class LintWithWorkspaceTest : NonProtobufGradlePluginLintTests, AbstractLintTest
     fun `lint with a config in workspace`() {
         assertSuccess()
     }
+
+    @Test
+    fun `lint a basic correct message with default config v2`() {
+        super.`lint a basic correct message with default config`()
```
Comment (andrewparmet) [inline src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: Heh. Do you have plans to hoist this up into the base test classes?

File: src/test/kotlin/build/buf/gradle/AbstractBufIntegrationTest.kt [hunk 2]
```diff
@@ -72,7 +72,7 @@ abstract class AbstractBufIntegrationTest : IntegrationTest {
                 "-PkotlinVersion=1.7.20",
                 "-PandroidGradleVersion=7.3.0",
             )
-            .withDebug(false) // Enable for interactive debugging
+            .withDebug(true) // Enable for interactive debugging
```
Comment (andrewparmet) [inline src/test/kotlin/build/buf/gradle/AbstractBufIntegrationTest.kt:75]: Does this affect anything if you don't want to debug? E.g. does it let tests run to completion or does it stop and wait for a debugger to attach?

File: src/test/resources/LintWithWorkspaceTest/lint_a_basic_correct_message_with_default_config_and_the_kotlin_android_plugin_v2/buf.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+version: v2
+modules:
+  - path: workspace
```
Comment (andrewparmet) [inline src/test/resources/LintWithWorkspaceTest/lint_a_basic_correct_message_with_default_config_and_the_kotlin_android_plugin_v2/buf.yaml:3]: For my own understanding - is `buf.work.yaml` just not a thing anymore in v2?

**Review by andrewparmet — APPROVED** [review]

File: src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt (hunk 1 above)
Comment (drice-buf) [reply src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: What would you suggest as a good idiom for two tests that do exactly the same thing but use different resource dirs?

Comment (drice-buf) [reply src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: I only added a `_v2` test for cases that contain a `buf.work.yaml` file, which are generally the `WithWorkspace` tests. Accordingly I define these cases only in the leaf classes.

File: src/test/resources/LintWithWorkspaceTest/lint_a_basic_correct_message_with_default_config_and_the_kotlin_android_plugin_v2/buf.yaml (hunk 3 above)
Comment (drice-buf) [reply src/test/resources/LintWithWorkspaceTest/lint_a_basic_correct_message_with_default_config_and_the_kotlin_android_plugin_v2/buf.yaml:3]: From `https://buf.build/docs/configuration/v1/buf-work-yaml`:

`Buf now has a v2 configuration available, where this file is no longer used. `

**Review by andrewparmet — APPROVED** [review]

File: src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt (hunk 1 above)
Comment (drice-buf) [reply src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: The `super` function is itself a `@Test` case that runs against the resource dir that does not end with `_v2`

Comment (andrewparmet) [reply src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: Right - and would it also make sense to run `v2` tests against a "vanilla" project (i.e., one not using workspaces)?

Comment (drice-buf) [reply src/test/kotlin/build/buf/gradle/LintWithWorkspaceTest.kt:27]: I don't think we need to test against every version of Buf CLI - apart from some edge cases which we unfortunately ran into, it is meant to be fully backwards compatible.
